osutil/exec.go

```diff
@@ -134,3 +138,101 @@ func CommandFromCore(name string, cmdArgs ...string) (*exec.Cmd, error) {
 	allArgs := append(ldSoArgs, cmdArgs...)
 	return exec.Command(coreLdSo, allArgs...), nil
 }
+
+var (
+	syscallKill    = syscall.Kill
+	syscallGetpgid = syscall.Getpgid
+
+	cmdWaitTimeout = 5 * time.Second
+)
+
+// KillProcessGroup kills the process group associated with the given command.
+//
+// If the command hasn't had Setpgid set in its SysProcAttr, you'll probably end
+// up killing yourself.
+func KillProcessGroup(cmd *exec.Cmd) error {
+	pgid, err := syscallGetpgid(cmd.Process.Pid)
+	if err != nil {
+		return err
+	}
+	if pgid == 1 {
+		return fmt.Errorf("cannot kill pgid 1")
+	}
+	return syscallKill(-pgid, syscall.SIGKILL)
+}
+
+// RunAndWait runs a command for the given argv with the given environ added to
+// os.Environ, killing it if it reaches timeout, or if the tomb is dying.
+func RunAndWait(argv []string, env []string, timeout time.Duration, tomb *tomb.Tomb) ([]byte, error) {
+	if len(argv) == 0 {
+		return nil, fmt.Errorf("cannot RunAndWait: need non-empty argv")
```

Comment:
do we have errors in this style?  I understand this are programmers error that should never happen, still wondering what's the usual style, I know we don't mention internal names in effectively user-visible errors